lib/melissa_data/web_smart/response_processor.rb

```diff
@@ -1,39 +1,40 @@
 module MelissaData
   module WebSmart
     module ResponseProcessor
-      def process_property(response)
-        codes = codes(response)
+      require 'yaml'
+      CODE_TYPES = [:property, :address]
 
+      def process(response, resp_type)
+        codes = codes(response, resp_type)
         if has_error_codes?(codes)
-          { errors: codes_for(codes, 'error') }
+          { errors: codes_for(codes, resp_type, 'error') }
         else
-          response.merge!(success: codes_for(codes, 'success'))
+          response.merge!(success: codes_for(codes, resp_type, 'success'))
         end
       end
 
-      def codes_for(codes, type)
-        codes.map { |c| method("property_#{type}_codes").call[c.to_sym] }.compact
+      def codes_for(codes, resp_type, code_type)
+        codes.map do |code|
+          YAML.load(File.read("config/#{resp_type}_#{code_type}_codes.yml")).values.first[code.to_s]
+        end.compact
       end
 
-      def codes(response)
-        response[:result][:code].split(",")
+      def codes(response, resp_type)
+        case resp_type
+        when 'property'
+          response[:result][:code].split(",")
+        when 'address'
+          response[:results].split(",")
+        end
```

Comment:
And lastly you would just have to write a rule for the case that will traverse to where the error codes are stored in the hash